### ISSUE TITLE
Build Error due to openssl 0.9.24

### ISSUE THREAD

**Issue by adrianbrink** [paritytech/substrate#886]
On macOS 10.14 (Mojave) running `cargo build` fails with

```bash
➜  substrate git:(master) cargo build
   Compiling openssl v0.9.24
   Compiling sr-io v0.1.0 (/Users/adrian/Desktop/code/polkadot/substrate/core/sr-io)
   Compiling safe-mix v1.0.0
   Compiling sr-sandbox v0.1.0 (/Users/adrian/Desktop/code/polkadot/substrate/core/sr-sandbox)
   Compiling trie-db v0.9.0 (https://github.com/paritytech/trie#b3b3209e)
   Compiling kvdb v0.1.0
   Compiling getset v0.0.6
   Compiling parking_lot v0.4.8
error: failed to run custom build command for `openssl v0.9.24`
process didn't exit successfully: `/Users/adrian/Desktop/code/polkadot/substrate/target/debug/build/openssl-6d9e1e02e1314785/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'Unable to detect OpenSSL version', /Users/adrian/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
error: build failed
```

This issue is also described in #800 . Both `openssl` and `libressl` are installed with `brew install openssl libressl`.

The `openssl` v0.9.24 and `openssl-sys` `v0.9.33` are pulled in through `ws` `0.7.8`. 

The two approaches I can currently think of are:
1. Overriding the `openssl` dependency with [patch](https://doc.rust-lang.org/cargo/reference/manifest.html#the-patch-section) or [replace](https://doc.rust-lang.org/cargo/reference/manifest.html#the-replace-section)
2. Upgrading `ws` to a later version.

**Comment by bkchr** [comment]
I looked into this and we are using already the latest version of `ws`. So, patching/replacing is currently the only way for solving this. :(

**Comment by gavofyork** [comment]
fixed.

**Comment by adrianbrink** [comment]
Confirmed the fix on macOS 10.14.

**Comment by bruciebruce** [comment]
Hi,

Excuse my green status is such matters.....

I cloned the substrate master on 10/30/18 and tried the following to get past the 'old' openssl dependency:

[patch.crates-io]
openssl = { git = 'https://github.com/kryptco/rust-openssl.git' }

but got this:

error: failed to run custom build command for `openssl v0.9.24`                                         
process didn't exit successfully: `/devel/substrate/target/debug/build/openssl-d6940fa3998f9474/build-script-build` (exit code: 101)
--- stderr thread 'main' panicked at 'Unable to detect OpenSSL version', /home/bruce/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

then I tried this:

[replace]
"openssl:0.9.23" = {git = 'https://github.com/kryptco/rust-openssl.git'}

but got this:

error: failed to run custom build command for `openssl v0.9.24`                                         
process didn't exit successfully: `/devel/substrate/target/debug/build/openssl-d6940fa3998f9474/build-script-build` (exit code: 101) --- stderr thread 'main' panicked at 'Unable to detect OpenSSL version', /home/bruce/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-0.9.24/build.rs:16:14
note: Run with `RUST_BACKTRACE=1` for a backtrace.

and then this:

[replace]
"openssl:0.9.24" = {git = 'https://github.com/kryptco/rust-openssl.git'}

but got this:

bruce@bruce-pc:/devel/substrate$ cargo build
 Updating git repository `https://github.com/kryptco/rust-openssl.git`
error: no matching package for override `https://github.com/rust-lang/crates.io-index#openssl:0.9.24` found location searched: https://github.com/kryptco/rust-openssl.git
version required: = 0.9.24

Here is my config:

bruce@bruce-pc:/devel/substrate$ uname -a
Linux bruce-pc 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

bruce@bruce-pc:~$ lsb_release -a 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic

I see people solving the above issue so I am at a loss.
Thank you in advance.
